emcc.py

```diff
@@ -111,7 +111,8 @@
 
 DEFAULT_ASYNCIFY_EXPORTS = [
   'main',
-  '__main_argc_argv'
+  '__main_argc_argv',
+  '_ZN10emscripten8internal5async*' # Embind's async functions.
```

Comment:
Relying on mangled named to find the exports seems hacky. Open to suggestions.

Reply:
Are there a fixed number of these?   Can we just use `extern 'C'` for them?   

How do these end up being called by name?

Reply:
These are template generated functions, so I don't think I can use extern c. Normal embind exported functions aren't actually exported but just put in the function table. It's a bit weird, but I use EMSCRIPTEN_KEEPALIVE on all the async embind template functions just so they have a name which I can then use to find them in the binaryen JSPI pass and convert them.

They aren't called by name but by function pointer.

Reply:
I see... I wonder if there is some other way we can flag these to the JSPI pass?    I guess maybe at least leave a comment here explaining that we don't actually need to export there at all?

Also, is this an embind feature that only works with ASYNCIFY=2 and not with ASYNCIFY=1?   Would it hard to make it work with both?

Reply:
> I see... I wonder if there is some other way we can flag these to the JSPI pass? 

Not that I can think of, but I'm open to suggestions.

> Also, is this an embind feature that only works with ASYNCIFY=2 and not with ASYNCIFY=1? Would it hard to make it work with both?

Yeah, this will really only work with `ASYNCIFY=2`. `ASYNCIFY=1` already kind of works with embind, since in `1` mode all indirect calls will be "asyncified" by default. I think we could make a mode where they behave the same, but after talking with Derek this seems low priority.